lib/tasks/dad_tasks.rake

```diff
@@ -265,7 +265,10 @@ namespace :dad_tasks do
                 " #{u.highest_level}.", User.where(is_admin: true).first)
     end
 
-    # STEP 9: Now that the daily job is complete, update.
+    # STEP 9: Process account deletions:
+    User.where(marked_for_deletion: true, deletion_date: Time.current.to_date).each(&:destroy)
```

Comment:
Almost! Use Time.now.utc.to_date... this makes the timer agnostic to the actual timezone of the server that runs this.